packages/api/src/utils/json-response.js

```diff
@@ -1,7 +1,13 @@
 /* eslint-env serviceworker */
+
+// TODO: move this to pin.js
+/**
+ * @typedef {{ error: { reason: string, details?: string } }} PinDataError
+ */
+
 export class JSONResponse extends Response {
   /**
-   * @param {BodyInit} body
+   * @param {BodyInit | PinDataError} body
```

Comment:
JSONResponse is used across all the endpoints. While having more strict parameter's types has its own advantages I wonder if keeping it looser just for now (with something like `{Object.<string, number>}`) would be a bit easier for now given the types are currently broken all over the place?

Don't have a strong opinion, just thinking aloud really.

Reply:
Do you mean adding `{Object.<string, number>}` instead of `PinDataError`? I'm ok with that, however it does give type error in `pins.js`

Reply:
sorry I meant 
```suggestion
   * @param {Object.<string, object>} body
```

Reply:
I mean {Object.<string, any>} however, I now realized that might not be the best as well. 
This might need more discussion. 

Shall we roll it back for now and tackle it in a separate issue?